lib/debt_to_income_ratio_latest/main.py

```diff
@@ -25,68 +25,75 @@ async def get_debt_to_income_ratio_latest(
 
     Returns:
         Ratio of debt / income for the given user. Returning 0 would correspond to a 0 debt amount,
-        and Inf would correspond to a 0 income amount.
+        and 'inf' would correspond to a 0 income amount.
     """
-
+    
+    # STEP 1: fetch list of institutions belonging to the user
     institutions = await api_client.institutions.get(user_uuid=user_uuid)
-    utc_starttime = utc_time - cutoff_interval
+    utc_starttime = utc_time - cutoff_interval    
 
     # subset to only fetch data for institutions known to contain loan-type accounts for the user
-    institutions_w_loan = [
-        inst for inst in institutions if "loan" in inst.account_types
-    ]
-
-    loan_inst_coroutines = [
-        api_client.balances.get(
-            user_uuid=user_uuid,
-            institution_id=institution.institution_id,
-            utc_starttime=utc_starttime,
-            utc_endtime=utc_time,
-            account_types=["loan"],
+    institutions_w_loan = []
+    for inst in institutions:
+        if "loan" in inst.account_types:
+            institutions_w_loan.append(inst)
```

Comment:
why use append here instead of keeping it how it was as a comprehension?

Reply:
Because comprehension structures are very tied to pythonic code, which is harder to translate to other languages in case that somebody wants to use the features as a reference to code their own in other languages.